### DISCUSSION TITLE
Windows?

### DISCUSSION THREAD

**Discussion by Travillion** [mkb79/Audible#347] [Q&A] [answered]
Good afternoon! I use audible and audible-cli a lot on my last machine, running it through the WSL. I just got a new Windows machine and I wonder if it's possible to just use in Windows? I saw there is a Windows install for audible-cli, but I don't see any specifics for audible.

Thank you

**Comment by mkb79** [comment]
Hi. Good morning. `Audible` and `audible-cli` are written in Python. So you only need a Python interpreter on your Windows machine. The you can use `pip` to install `audible-cli`. If you download the windows binary from the [releases](https://github.com/mkb79/audible-cli/releases) pages, it will bundle all dependencies. If you use the single-file version, the start-up time on Windows is high. If this is to much for you, please use the dir version.

**Comment by Travillion** [comment]
Thank you! Is there any functional difference between running in windows
vs. WSL? You mentioned the high startup time for the single file version.
Any other considerations? I could do WSL if the scripts run better there,
I'm just trying to streamline things this time around.

**Comment by mkb79** [comment]
There should no functional differences, if you run it on Windows natively. But for a better performance I would suggest to install `audible-cli` using [pipx](https://github.com/pypa/pipx). So you get around additional startup time caused by creating the binary with Pyinstaller.

**Comment by Travillion** [comment]
Oh great, thank you! Is it just cli that I should use pipx for, or also
audible?

**Reply by mkb79** [reply]
Short form:
It's Python related. Pipx manages the dependencies for `audible-cli` in its own virtual environment. This way you can use my package independent from other Python apps you're installing.

**Reply by Travillion** [reply]
I installed Audible via pip install and I installed audible-cli via pipx. When I try to run audible-quickstart I get an error 
`PS C:\MediaTools> audible-quickstart
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "c:\users\travis\.local\bin\audible-quickstart.exe\__main__.py", line 4, in <module>
  File "C:\Users\Travis\AppData\Local\pipx\pipx\venvs\audible-cli\Lib\site-packages\audible_cli\__init__.py", line 5, in <module>
    from .cli import main, quickstart
  File "C:\Users\Travis\AppData\Local\pipx\pipx\venvs\audible-cli\Lib\site-packages\audible_cli\cli.py", line 3, in <module>
    from pkg_resources import iter_entry_points
ModuleNotFoundError: No module named 'pkg_resources'`

Any idea why this is not running?

**Reply by mkb79** [reply]
`pkg_resources` should be installed by default. Now you can run `pipx inject audible-cli pkg_resources`. This should solve your issue. 

FYI:
If you install audible-cli via pipx, you don’t need to install audible via pip. The first is enough.

**Comment by Travillion** [comment]
Thank you for your quick response! I tried the inject command and got this
error:

"PS C:\MediaTools> pipx inject audible-cli pkg_resources
Fatal error from pip prevented installation. Full pip output in file:

C:\Users\Travis\AppData\Local\pipx\pipx\Logs\cmd_2024-02-19_13.00.58_pip_errors.log

Some possibly relevant errors from pip install:
    ERROR: Could not find a version that satisfies the requirement
pkg-resources (from versions: none)
    ERROR: No matching distribution found for pkg-resources
Error installing pkg-resources."

I used pip uninstall audible and pipx uninstall audible-cli, then pipx
install audible-cli, to try and start fresh, and I got the same behavior
(unable to run audible-quickstart, unable to inject pkg-resources).
Examining the error log and it only displays the two mentioned above.

**Reply by mkb79** [reply, marked as answer]
Oh, my mistake. Can you try `pipx inject audible-cli setuptools`?!

**Comment by Travillion** [comment]
That worked, thank you!!
